### ISSUE TITLE
Custom (hand made) HTML completely mangled

### ISSUE THREAD

**Issue by timreeves** [WordPress/gutenberg#7791]
**Describe the bug**
In particular, before Gutenberg I was using a plugin to suppress the standard WordPress formatting, to get raw HTML. Now, Gutenberg does not honour that previous choice and various newlines in my code are converted to breaks, which completely messes the page layout. This happens both in the page output and on page editing - when I select the "Raw HTML" editing option in Gutenberg, the original newlines are missing, so I can hardly read the code; and empty inline anchor links in this form are completely missing:
```
<a class="lpbox lpbox3" href="#"></a>
```
Converting Newlines to break tags unasked breaks various layouts, so this is an absolute no go. I find it simply disrespectful - Gutenberg does NOT know what I want or need better than I do! At the very least, it should add an option to allow REAL raw coding, in which NOTHING gets changed.

**To Reproduce**
Steps to reproduce the behavior: Add HTML code (you'll probably need to do it without Gutenberg) exactly like this:
```
<div id="lpsizebox">
<img id="size-helper" src="/wp-content/uploads/images/lpsizer.png" width="100" height="68" alt="">
<div id="lpcontent">
<a class="lpbox lpbox3" href="#"></a>
<a class="lpbox lpbox1" href="#"></a>
<a class="lpbox lpbox6" href="#"></a>
<a class="lpbox lpbox8" href="#"></a>
<a class="lpbox lpbox4" href="#"></a>
<a class="lpbox lpbox7" href="#"></a>
<a class="lpbox lpbox5" href="#"></a>
<a class="lpbox lpbox2" href="#"></a>
<a class="lpenterlink" href="/home"><img id="lpweiter" src="/wp-content/uploads/images/weiter_d.png" width="155" height="35" alt=""></a>
</div>
<a class="lplanguage" rel="alternate" href="https://werteagentur.net/en/"><img src="https://werteagentur.net/wp-content/uploads/images/flags/United-Kingdom.png" alt="English">English</a>
<a class="lphomelink" href="/home"></a>
</div>
<footer class="lpfooter">
<a id="lpfooterfirst" href="/impressum">Impressum</a>
<a href="/kontakt">Kontakt</a>
<a href="/datenschutz">Datenschutz</a>
<a href="/sitemap">Sitemap</a>
</footer>
```
**Expected behavior**
The code should not be mangled in any way - I need an option to get the classic editor on a per page basis. BTW This is my new landing page in development.

**Actual behaviour**
This is what the Gutenberg Raw HTML shows:
```
<div id="lpsizebox"><img id="size-helper" src="/wp-content/uploads/images/lpsizer.png" alt="" width="100" height="68" /><br /><br /><br /><br /><br /><br /><br /><br /><br /><a class="lplanguage" href="https://werteagentur.net/en/" rel="alternate"><img src="https://werteagentur.net/wp-content/uploads/images/flags/United-Kingdom.png" alt="English" />English</a><br /><br /><a class="lpenterlink" href="/home"><img id="lpweiter" src="/wp-content/uploads/images/weiter_d.png" alt="" width="155" height="35" /></a></div>
<footer class="lpfooter"><a id="lpfooterfirst" href="/impressum">Impressum</a><br /><a href="/kontakt">Kontakt</a><br /><a href="/datenschutz">Datenschutz</a><br /><a href="/sitemap">Sitemap</a></footer>
```

**Desktop (please complete the following information):**
 - OS: Kubuntu
 - Browser Firefox
 - Version 61.0

**Additional context**
- Gutenberg 3.2.0

**Comment by pojke** [comment]
@timreeves Tested in Gutenberg 3.2.0 under raw HTML block and it works as expected.
Any other block than raw HTML one should not be used for raw HTML.
(Not mentioning that HTML in post content is not advisable in the first place)

**Comment by timreeves** [comment]
OK, I'll take that on trust as it sounds reasonable. I can't test it directly having deleted the Gutenberg plugin.

But let's reflect on what happened: I installed Gutenberg on an existing website - this is effectively what will happen worldwide without explicit user action if Gutenberg is indeed distributed in WP5 core. So my hand-coded landing page was already there, pre-Gutenberg. I'm using the Weaver Xtreme theme which has a per-page option of its own to suppress WP texturizing etc. and I had set this on. So even if Gutenberg checks for one of the known plugins (e.g. Raw HTML or WP Edit with the wp-autop option on) you would not find it. There must be so many ways to suppress the WP processing that the only reliable way to find out would be to compare the original and final text and see if it has been done.

So what actually happened is that Gutenberg found an existing page and classified it as one normal text block. This was the error, it should have been classified as a Raw HTML type of block.

A question: Can I change the type of an existing block in Gutenberg?

@pojke I disagree strongly that HTML is not advisable in post/page content. I'm a programmer and my colleague is a great media designer, I let her do her thing and then it's my job to get it implemented. This will sometimes involve a "special" like a landing page, or the choice of WP Bakery Pagebuilder, or Intense Shortcodes or Shortcodes Ultimate for all pages to get the features we need. I've even been known to manually add a flexbox to a page when it was needed. And I maintain the [TinyMCE Custom Styles](https://wordpress.org/plugins/tinymce-custom-styles/) plugin which allows to define and use own styles within TinyMCE. It clocks over 7.000 downloads, so obviously I'm not the only one doing hands-on work in WordPress.

The bottom line here is for me: When Gutenberg is installed in an existing website it must very intelligently decide, individually for each page, whether to categize it as Raw HTML or normal text.

**Comment by timreeves** [comment]
Thanks @pojke 

If I understand issue #7466 right, it's alluding to converting one classic block (TinyMCE MonoBlock) into individual Gutenberg blocks by type. Which would allow them to be further formatted and sorted using Gutenberg. And the author is saying why not provide that feature for HTML Blocks as well. I personally would'nt want to do that, I reckon, but if it's just an option then well maybe some people will want it.

Take my example above: Gutenberg did not (yet) notice the actual type of the content and made a conversion to a wrong type. Thinking it was a classic block, it added breaks which broke the layout, and completely deleted the empty links which it contains - which are styled in CSS and do a great job - they are not empty on the screen!

Re Page Templates: The site in question uses the (fantastic) Weaver Xtreme theme, which provides 8 (!) page templates of its own. I had selected the correct one - "blank" - and the "raw html" option for my landing page. Gutenberg missed both settings and just "did its thing". My site will only run properly with a plugin to disable Gutenberg, if it turns up like this in WordPress core.

Seriously, that's not the best way to get likes :) I was just speaking to another professional and she said, oh it's terrible now even open source software like WordPress is becoming like Microsoft - they assume the lowest common denominator and put everybody - including professionals who could and would do more - into the same straight jacket...

On a conciliatory note: I see a lot of good and hard work going into Gutenberg, and appreciate the quick responses here. I also now understand that I can still "do my thing" with Gutenberg if I define a raw html block. Where I see the big problem is forcing Gutenberg on everyone by putting it in the core, and then it breaking a huge number of existing websites. To avoid this I vociferously suggest a lot of work going into the question of recognising whether an exisitng page (seen for the first time) should be classified as a classic or raw block. That, basically, was my problem.

One last question: What's the relationship between Gutenberg and WP Bakery Pagebuilder (aka Visual Composer)? Will I need a "Disable Gutenberg" plugin, or does VC do that itself, or do they impede each other... the mind boggles :)

**Comment by pojke** [comment]
@timreeves The conversation sidetracks from the Gutenberg bug itself. If you have any questions relating to how to develop in WordPress, use custom page templates and avoid using HTML in page content, join your local community on Meetup, slack or find the right channel on WordPress main slack https://make.wordpress.org/chat/

The solution to your problem should be converting paragraph block into html blocks. It has to be done manually, there's no easy way for the software to, without a doubt, recognise basic editor html using paragraphs and images from complex html that you're using and safely select the right block. 

I suggest we close this issue.

**Comment by timreeves** [comment]
I agree that this has gone off topic - have opened feature request #7859.
